### PR TITLE
Initial project

### PR DESCRIPTION
**Pull request by nandadubey**
Please review and DO NOT merge as this branch is only created to create review request and get feedback.

### CODE REVIEW

File: affirm/src/main/java/com/affirm/android/AffirmPlugins.java [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+            clientBuilder.readTimeout(30, TimeUnit.SECONDS);
+            clientBuilder.followRedirects(false);
+
+            mRestClient = AffirmHttpClient.createClient(enableTls12OnPreLollipop(clientBuilder));
+        }
+        return mRestClient;
+    }
+
+    /**
+     * Enables TLS v1.2 when creating SSLSockets.
```
Comment (hectorups) [inline affirm/src/main/java/com/affirm/android/AffirmPlugins.java]: As mentioned before, this is not necessary when we bump up the min sdk version to 21

File: affirm/build.gradle [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+ext {
+    BUILD_TOOLS_VERSION = "28.0.3"
+    COMPILE_SDK_VERSION = 28
+
+    androidxAppCompatVersion = '1.0.2'
+    junitVersion = "4.12"
+    testRunnerVersion = "1.0.2"
+    testEspressoVersion = "3.0.2"
+    okhttpVersion = '3.12.1'
```
Comment (hectorups) [inline affirm/build.gradle]: set the latest stable 3.14.1

File: affirm/src/main/java/com/affirm/android/AffirmResponseWrapper.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.affirm.android;
+
+import com.affirm.android.exception.AffirmException;
+
+class AffirmResponseWrapper<T> {
+    T source;
```
Comment (hectorups) [inline affirm/src/main/java/com/affirm/android/AffirmResponseWrapper.java]: set these 2 fields as final?

File: affirm/src/main/java/com/affirm/android/AffirmHttpClient.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+                AffirmTracker.track(NETWORK_ERROR, ERROR,
+                        createTrackingNetworkJsonObj(okHttpRequest, response));
+            }
+
+            final Headers headers = response.headers();
+            String requestId = headers.get(X_AFFIRM_REQUEST_ID);
+            if (response.code() < 200 || response.code() >= 300) {
+                ResponseBody responseBody = response.body();
+                final AffirmError affirmError = AffirmPlugins.get().gson()
+                        .fromJson(responseBody != null
```
Comment (hectorups) [inline affirm/src/main/java/com/affirm/android/AffirmHttpClient.java]: Use fromJson(Reader) not fromJson(String) for performance. You can use response.body().contentLength() == 0L to see if it's empty

Sadly, the servers might return html. For example on 500s. I would be ok with ignoring json AffirmError serializing errors

File: affirm/build.gradle [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    mockitoCoreVersion = '2.21.0'
+    robolectricVersion = '4.1'
+}
+
+android {
+    compileSdkVersion COMPILE_SDK_VERSION
+    buildToolsVersion BUILD_TOOLS_VERSION
+
+    defaultConfig {
+        minSdkVersion 14
```
[the remark this answers is not in the document]
Comment (nandadubey) [reply affirm/build.gradle]: Closing this PR as these issues are now fixed on master branch

File: affirm/src/main/java/com/affirm/android/AffirmHttpClient.java (hunk 4 above)
Comment (nandadubey) [reply affirm/src/main/java/com/affirm/android/AffirmHttpClient.java]: Closing this PR as these issues are now fixed on master branch

**Pull request closed without merging**